src/lib/wicked/adapter.js

```diff
@@ -113,6 +113,35 @@ class WickedAdapter {
         return result;
     }
 
+
+    /*
+     * Obtains the DNS global settings from /etc/sysconfig/network/config
+     */
+    async dnsSettings() {
+        const filePath = `/etc/sysconfig/network/config`;
+        const file = await new SysconfigFile(filePath).read();
+        const policy = await file.get("NETCONFIG_DNS_POLICY") || "";
+        const nameServers = await file.get("NETCONFIG_DNS_STATIC_SERVERS") || "";
+        const searchList = await file.get("NETCONFIG_DNS_STATIC_SEARCHLIST") || "";
+
+        return { policy, nameServers, searchList };
```

Comment:
For connections, interfaces, and routes, we have some factories (like `createInterface`) that are useful to normalize objects like this one.